sdk/go/common/resource/plugin/langruntime.go

```diff
@@ -16,22 +16,80 @@ package plugin
 
 import (
 	"context"
+	"fmt"
 	"io"
 
 	"github.com/blang/semver"
 	"github.com/hashicorp/hcl/v2"
 	"github.com/pulumi/pulumi/sdk/v3/go/common/resource"
 	"github.com/pulumi/pulumi/sdk/v3/go/common/resource/config"
 	"github.com/pulumi/pulumi/sdk/v3/go/common/workspace"
+	pulumirpc "github.com/pulumi/pulumi/sdk/v3/proto/go"
+	structpb "google.golang.org/protobuf/types/known/structpb"
 )
 
+// ProgramInfo contains minimal information about the program to be run.
+type ProgramInfo struct {
+	root       string
+	program    string
+	entryPoint string
+	options    map[string]any
+}
+
+func NewProgramInfo(rootDirectory, programDirectory, entryPoint string, options map[string]any) ProgramInfo {
+	return ProgramInfo{
+		root:       rootDirectory,
+		program:    programDirectory,
+		entryPoint: entryPoint,
+		options:    options,
+	}
+}
+
+// The programs root directory, i.e. where the Pulumi.yaml file is.
+func (info ProgramInfo) RootDirectory() string {
+	return info.root
+}
+
+// The programs directory, generally the same or a subdirectory of the root directory.
+func (info ProgramInfo) ProgramDirectory() string {
+	return info.program
+}
+
+// The programs main entrypoint, i.e. the path to the main program file relative to program directory.
+func (info ProgramInfo) EntryPoint() string {
+	return info.entryPoint
+}
+
+// Execution runtime options for the program.
```

Comment:
Not sure "Execution" is the right term for these options. I'd probably just leave it as "Runtime plugin options for the program"